framework/src/media/InputHandler.cpp

```diff
@@ -71,12 +72,118 @@ bool InputHandler::doStandBy()
 	return true;
 }
 
+bool InputHandler::open()
```

Comment:
How about apply polymorphism?
open() for prepare
open(something) for prepareAsync or change open(void) to open(bool isSync)
But in my opinion, you have to set policy first. below is my opinion.
1. prepare
 a. Download amount of data to check type of contents
 b. do it until it gets correct info of header (until it reaches threshold)
 c. Also add description in prepare like this.
   "prepare is sync call, so if it takes a lot of time until it ready, please use prepareAsync"

2. prepareAsync
  a. same operation as above 'a' and 'b' but thread should be there.
  b. also add more description there.

Reply:
I moved common logic to StreamHandler::open(), and override open() function in InputHandler to add buffering sync logic.